bridgeql/django/models.py

```diff
@@ -216,45 +204,40 @@ def query_has_properties(self):
         # TODO show error if distinct is True and properties are present in fields
         if self.params.distinct:
             return False
-        return bool(set(self.params.fields).intersection(self.model_config.get_properties()))
+        return bool(set(self.params.fields).intersection(
+            self.model_config.get_properties()))
 
-    def _add_fields(self):
+    def _add_properties(self, db_rows, query_properties):
+        # evaluate queryset values
+        db_rows = list(db_rows)
         qset_values = DBRows()
-        self.qset = self.qset.select_related()
-        logger.debug('Request parameters: %s \nQuery: %s\n',
+        # skipping select_related call, expecting
+        # properties will not have references call
+        # like machine.os.arch in any of the machine's property
+        logger.debug('Request parameters: %s \nQuery (2nd call): %s\n',
                      self.params.params, self.qset.query)
-        for row in self.qset:
-            model_fields = {}
-            for field in self.params.fields:
-                attr = row
-                for ref in field.split('__'):
-                    try:
-                        attr = getattr(attr, ref)
-                        if attr is None:
-                            break
-                    except AttributeError:
-                        raise InvalidModelFieldName(
-                            'Invalid query for field %s in %s.' % (ref, attr))
-                model_fields[field] = attr
-            qset_values.append(model_fields)
+        for i, row in enumerate(self.qset):
```

Comment:
We realised quickly with a load that this 2nd query was making data integrity issue thus we had to revert use of another query for optimization.
Fixed the slowness of getattr by passing correct list of references in select_related().